rclpy/rclpy/client.py

```diff
@@ -122,8 +122,8 @@ def call_async(self, request: SrvTypeRequest) -> Future:
         if not isinstance(request, self.srv_type.Request):
             raise TypeError()
 
-        with self.handle as capsule:
-            sequence_number = _rclpy.rclpy_send_request(capsule, request)
+        with self.handle:
+            sequence_number = self.__client.send_request(request)
```

Comment:
@sloretz nit^N: perhaps using the outcome of `self.handle.__enter__` is easier to read (though equivalent)?

```suggestion
        with self.handle as client:
            sequence_number = client.send_request(request)
```
Same elsewhere.

Reply:
I went the opposite direction in a6149d10cc9120bf582381698c31f1d137efc296 and made the context manager return `None` when it's entered. I'm a little cautions because I don't know if pybind11 has a way to figure out the `Destroyable` type returned at `__enter__` is an instance of `rclpy::Client`. If `__enter__` returns `Destroyable d`, then is `client.send_request`  effectively `reinterpret_cast<Client *>(d)->send_request()`? I think I recall @mxgrey showing me a case (IIRC involving multiple inheritance) where that's unsafe to do.

Reply:
Aha! Interesting. Good to know.